app/scripts/modules/core/src/presentation/forms/inputs/NumberInput.tsx

```diff
@@ -9,10 +11,37 @@ interface INumberInputProps extends IFormInputProps, React.InputHTMLAttributes<a
   inputClassName?: string;
 }
 
-export const NumberInput = (props: INumberInputProps) => {
-  const { field, validation, inputClassName, ...otherProps } = props;
-  const fieldProps = { ...field, value: orEmptyString(field.value) };
-  const className = `NumberInput form-control ${orEmptyString(inputClassName)} ${validationClassName(validation)}`;
+const isNumber = (val: any): val is number => typeof val === 'number';
 
-  return <input className={className} type="number" {...fieldProps} {...otherProps} />;
-};
+export class NumberInput extends React.Component<INumberInputProps> {
+  private getMinValidator = memoizeOne((min: any) => (isNumber(min) ? Validation.minValue(min) : undefined));
+  private getMaxValidator = memoizeOne((max: any) => (isNumber(max) ? Validation.maxValue(max) : undefined));
+
+  private validator: Validator = (val: any, label?: string) => {
+    const { min, max } = this.props;
+    const validators = [this.getMinValidator(min), this.getMaxValidator(max)].filter(x => !!x);
+
+    if (validators.length === 1) {
+      return validators[0](val, label);
```

Comment:
Can `composeValidators` just handle this case?

Reply:
Yes! great idea, thanks.  I wanted to optimize this since this gets called a lot, but the optimization definitely belongs in `composeValidators`

Reply:
Moved into `composeValidators` and this feels much better now, thanks.